pkg/globalnet/controllers/cluster_egressip_controller.go

```diff
@@ -128,14 +145,177 @@ func (c *clusterGlobalEgressIPController) onCreate(egressIP *submarinerv1.Cluste
 				ClusterGlobalEgressIPName),
 		})
 
+		return errors.Errorf("ClusterGlobalEgressIP with name %q is not supported, only well-known"+
+			" name %q is supported", egressIP.Name, ClusterGlobalEgressIPName)
+	}
+
+	if *numberOfIPs < 0 {
+		tryAppendStatusCondition(&egressIP.Status, &metav1.Condition{
+			Type:    string(submarinerv1.GlobalEgressIPAllocated),
+			Status:  metav1.ConditionFalse,
+			Reason:  "InvalidInput",
+			Message: "The NumberOfIPs cannot be negative",
+		})
+
+		return errors.Errorf("NumberOfIPs %q in %q cannot be less than 0", *numberOfIPs, egressIP.Name)
+	}
+
+	if *numberOfIPs == 0 {
+		tryAppendStatusCondition(&egressIP.Status, &metav1.Condition{
+			Type:    string(submarinerv1.GlobalEgressIPAllocated),
+			Status:  metav1.ConditionFalse,
+			Reason:  "ZeroInput",
+			Message: "No global IPs to allocate",
+		})
+
+		return errors.Errorf("NumberOfIPs %q in %q cannot be 0", *numberOfIPs, egressIP.Name)
+	}
+
+	return nil
+}
+
+func (c *clusterGlobalEgressIPController) onCreate(key string, numberOfIPs *int, status *submarinerv1.GlobalEgressIPStatus) bool {
+	// When Globalnet Pod is restarted or migrated, we just have to Sync the IPPool cache and IPTable rules.
+	if *numberOfIPs == len(status.AllocatedIPs) {
+		err := c.pool.Reserve(status.AllocatedIPs...)
+		if err != nil {
+			klog.Errorf("Error allocating IPs %v for %q: %v", status.AllocatedIPs, key, err)
+			tryAppendStatusCondition(status, &metav1.Condition{
+				Type:    string(submarinerv1.GlobalEgressIPAllocated),
+				Status:  metav1.ConditionFalse,
+				Reason:  "IPPoolAllocationFailed",
+				Message: fmt.Sprintf("Error allocating %v global IP(s) from the pool: %v", status.AllocatedIPs, err),
+			})
+
+			return false
+		}
+
+		if err := c.programClusterGlobalEgressRules(status); err != nil {
+			_ = c.pool.Release(status.AllocatedIPs...)
+			status.AllocatedIPs = []string{}
+
+			klog.Errorf("Error syncing the IPTable rules on the node for %q: %v", key, err)
+
+			return true
```

Comment:
I wonder if at some point we should make those entitites separate:
1) The one handling allocations & reporting to the object
2) The one configuring egress rules, because it listenst to  GlobalEgressIPs